static/app/components/workflowEngine/form/control/priorityControl.tsx

```diff
@@ -0,0 +1,218 @@
+import {useCallback, useEffect, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {GroupPriorityBadge} from 'sentry/components/badge/groupPriority';
+import {Chevron} from 'sentry/components/chevron';
+import {CompactSelect, type SelectOption} from 'sentry/components/compactSelect';
+import {Flex} from 'sentry/components/container/flex';
+import FieldWrapper from 'sentry/components/forms/fieldGroup/fieldWrapper';
+import NumberField from 'sentry/components/forms/fields/numberField';
+import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {IconArrow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import {PriorityLevel} from 'sentry/types/group';
+
+export interface PriorityControlGridProps {
+  name: string;
+  onChange?: (value: PriorityThresholds) => void;
+  value?: PriorityThresholds;
+}
+
+export interface PriorityThresholds {
+  created: PriorityLevel;
+  high?: number;
+  medium?: number;
+}
+
+export default function PriorityControl({
+  name,
+  value,
+  onChange,
+}: PriorityControlGridProps) {
+  const [thresholds, setThresholds] = useState<PriorityThresholds>(
+    value ?? {
+      created: PriorityLevel.LOW,
+      [PriorityLevel.MEDIUM]: 0,
+      [PriorityLevel.HIGH]: 0,
+    }
+  );
```

Comment:
this data structure feels a little off to me - it feels like we should have a property `priority` that would set the priority level for the created input field, then we can make this a consistent data structure that just sets values based on the priority level.

it's feeling a bit confusing to try and add all of this data into a single state object. it almost feels like there should be an internal context that uses the priority and can be accessed at different layers so we can reduce re-render size on state change. (especially since this will need to update state on keystroke)